### ISSUE TITLE
Losing scroll events on Android

### ISSUE THREAD

**Issue by jcf120** [shortercode/Doom.js#13]
My Android touch hot fix #1 has come back to haunt me. (Upgraded web timing and now it effectively locks up if you scroll down too far.) I'm gonna have a go at reviving your global touch model. Suggestions welcome!

**Comment by jcf120** [comment]
Well, I got the global touch model up and running. Unfortunately it doesn't fix the original problem it was meant to solve. I'm tempted to just revert MR #1. (In my case it's now creating many more problems than it solves.)

**Comment by shortercode** [comment]
All solutions are problems waiting to happen ;) Was a bit of a long shot hoping it would fix it I guess, but it's a shame it doesn't offer other gains. If it's more of a hindrance then strip it out.

As for other solutions... Not sure there's any hard and fast one's. I believe hammer js gets past the issue by blocking scrolling by default and allowing the developer to re-enable it via a flag for horizontal and/or vertical motion. This would solve it for most situations, although when I wrote it originally it was more of a "do as little as possible, and let the developer deal with the event as they see fit".

Another alternative is to bundle a hand coded scroll, which is definitely possible but will change the device "feel". As well as requiring us to correctly apply momentum, which can be a bit of a pain.

**Comment by jcf120** [comment]
Yeah, I happy explicitly handling it case by case. I suppose on the one view where it was originally a problem I can just call `preventDefault` using the `ontap` callback.

**Comment by shortercode** [comment]
All solutions are problems waiting to happen ;) Was a bit of a long shot
hoping it would fix it I guess, but it's a shame it doesn't offer other
gains. If it's more of a hindrance then strip it out.

As for other solutions... Not sure there's any hard and fast one's. I
believe hammer js gets past the issue by blocking scrolling by default and
allowing the developer to reenable it via a flag for horizontal and/or
vertical motion. This would solve it for most situations, although when I
wrote it originally it was more of a "do as little as possible, and let the
developer deal with the event as they see fit".

The final alternative is to bundle a hand coded scroll, which is definitely
possible but will adjust the device "feel". As well as requiring us to
correctly apply momentum. I've avoided that as the scrolling graph in web
timing was once like that (albeit using a buffer canvas and offsetting the
texture) but I seemed to get erroneous timestamps on certain devices and
could never figure out what caused it. So I rewrote the thing, made my life
much easier!
